x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -363,35 +380,44 @@ private void getVersionedDocuments(Collection<String> uids, ActionListener<Tuple
-                                        itemResponse.getFailure().getFailure()
-                                    );
-                                }
+                                errors.put(profileUid, itemResponse.getFailure().getFailure());
                             } else if (itemResponse.getResponse() != null) {
                                 if (itemResponse.getResponse().isExists()) {
-                                    retrievedDocs.add(buildProfileDocument(itemResponse.getResponse().getSourceAsBytesRef()));
-                                } else if (logger.isDebugEnabled()) {
-                                    logger.debug("Profile [{}] not found", docIdToUid(itemResponse.getId()));
+                                    retrievedDocs.add(
+                                        new VersionedDocument(
+                                            buildProfileDocument(itemResponse.getResponse().getSourceAsBytesRef()),
+                                            itemResponse.getResponse().getPrimaryTerm(),
+                                            itemResponse.getResponse().getSeqNo()
+                                        )
+                                    );
+                                } else {
+                                    logger.debug("Profile [{}] not found", profileUid);
+                                    errors.put(profileUid, new ResourceNotFoundException("profile document not found"));
                                 }
                             } else {
                                 assert false
                                     : "Inconsistent mget item response [" + itemResponse.getIndex() + "] [" + itemResponse.getId() + "]";
                                 logger.error("Inconsistent mget item response [{}] [{}]", itemResponse.getIndex(), itemResponse.getId());
                             }
                         }
-                        if (loggedException != null) {
-                            logger.debug(() -> format("Failed to retrieve profiles %s", failures), loggedException);
+                        final ResultsAndErrors<VersionedDocument> resultsAndErrors = new ResultsAndErrors<>(retrievedDocs, errors);
+                        if (logger.isDebugEnabled() && false == resultsAndErrors.errors().isEmpty()) {
+                            Exception loggedException = null;
+                            final List<String> errorUids = errorUidsExcludingNotFound(resultsAndErrors.errors());
+                            for (String uid : errorUids) {
+                                loggedException = ExceptionsHelper.useOrSuppress(loggedException, resultsAndErrors.errors().get(uid));
+                            }
+                            if (loggedException != null) {
+                                logger.debug(() -> format("Failed to retrieve profiles %s", errorUids), loggedException);
+                            }
```

Comment:
I have a change of heart on how errors are handled in multi-get. I am now in favor of reporting all errors in detail and categorize not found uids as error as well. This errorUids extraction is added to maintain the existing behaviour. But if you are happy with the new approach, I can raise a follow-up PR to have errors reported in details for the hasPrivileges check.

Reply:
I don't think the logging as a whole makes sense anymore since all the `errors` are returned in the response now.

Reply:
Agreed. I'll remove it in the follow-up PR for adding the `errors` field in the profileHasPrivileges response.